### PR TITLE
Add page types for Add-ons

### PR DESCRIPTION
**Pull request by wbamberg**
Add page types for Add-ons, per https://github.com/mdn/content/issues/23942.

cc @rebloor :)

### CODE REVIEW

File: files/en-us/mozilla/add-ons/webextensions/api/contentscripts/registeredcontentscript/unregister/index.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ---
 title: contentScripts.RegisteredContentScript.unregister()
 slug: >-
+page-type: webextension-api-function
   Mozilla/Add-ons/WebExtensions/API/contentScripts/RegisteredContentScript/unregister
```
Comment (wbamberg) [inline files/en-us/mozilla/add-ons/webextensions/api/contentscripts/registeredcontentscript/unregister/index.md:5]:
```suggestion
slug: >-
  Mozilla/Add-ons/WebExtensions/API/contentScripts/RegisteredContentScript/unregister
page-type: webextension-api-function
```

File: files/en-us/mozilla/add-ons/webextensions/api/devtools/panels/elementspanel/createsidebarpane/index.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ---
 title: devtools.panels.ElementsPanel.createSidebarPane()
 slug: >-
+page-type: webextension-api-function
   Mozilla/Add-ons/WebExtensions/API/devtools/panels/ElementsPanel/createSidebarPane
```
Comment (wbamberg) [inline files/en-us/mozilla/add-ons/webextensions/api/devtools/panels/elementspanel/createsidebarpane/index.md:5]:
```suggestion
  Mozilla/Add-ons/WebExtensions/API/devtools/panels/ElementsPanel/createSidebarPane
page-type: webextension-api-function
```

File: files/en-us/mozilla/add-ons/webextensions/api/devtools/panels/elementspanel/onselectionchanged/index.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ---
 title: onSelectionChanged
 slug: >-
+page-type: webextension-api-event
   Mozilla/Add-ons/WebExtensions/API/devtools/panels/ElementsPanel/onSelectionChanged
```
Comment (wbamberg) [inline files/en-us/mozilla/add-ons/webextensions/api/devtools/panels/elementspanel/onselectionchanged/index.md:5]:
```suggestion
  Mozilla/Add-ons/WebExtensions/API/devtools/panels/ElementsPanel/onSelectionChanged
page-type: webextension-api-event
```

File: files/en-us/mozilla/add-ons/webextensions/api/devtools/panels/extensionsidebarpane/onhidden/index.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 ---
 title: devtools.panels.ExtensionSidebarPane.onHidden
 slug: >-
+page-type: webextension-api-event
   Mozilla/Add-ons/WebExtensions/API/devtools/panels/ExtensionSidebarPane/onHidden
```
Comment (wbamberg) [inline files/en-us/mozilla/add-ons/webextensions/api/devtools/panels/extensionsidebarpane/onhidden/index.md:5]:
```suggestion
  Mozilla/Add-ons/WebExtensions/API/devtools/panels/ExtensionSidebarPane/onHidden
page-type: webextension-api-event
```

File: files/en-us/mozilla/add-ons/webextensions/api/devtools/panels/extensionsidebarpane/setobject/index.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 ---
 title: devtools.panels.ExtensionSidebarPane.setObject()
 slug: >-
+page-type: webextension-api-function
   Mozilla/Add-ons/WebExtensions/API/devtools/panels/ExtensionSidebarPane/setObject
```
Comment (wbamberg) [inline files/en-us/mozilla/add-ons/webextensions/api/devtools/panels/extensionsidebarpane/setobject/index.md:5]:
```suggestion
  Mozilla/Add-ons/WebExtensions/API/devtools/panels/ExtensionSidebarPane/setObject
page-type: webextension-api-function
```

File: files/en-us/mozilla/add-ons/webextensions/api/webrequest/max_handler_behavior_changed_calls_per_10_minutes/index.md [hunk 6]
```diff
@@ -1,6 +1,7 @@
 ---
 title: webRequest.MAX_HANDLER_BEHAVIOR_CHANGED_CALLS_PER_10_MINUTES
 slug: >-
+page-type: webextension-api-property
   Mozilla/Add-ons/WebExtensions/API/webRequest/MAX_HANDLER_BEHAVIOR_CHANGED_CALLS_PER_10_MINUTES
```
Comment (wbamberg) [inline files/en-us/mozilla/add-ons/webextensions/api/webrequest/max_handler_behavior_changed_calls_per_10_minutes/index.md:5]:
```suggestion
  Mozilla/Add-ons/WebExtensions/API/webRequest/MAX_HANDLER_BEHAVIOR_CHANGED_CALLS_PER_10_MINUTES
page-type: webextension-api-property
```

File: files/en-us/mozilla/add-ons/webextensions/api/devtools/panels/extensionsidebarpane/setexpression/index.md [hunk 7]
```diff
@@ -1,6 +1,7 @@
 ---
 title: devtools.panels.ElementsPanel.setExpression()
 slug: >-
+page-type: webextension-api-function
   Mozilla/Add-ons/WebExtensions/API/devtools/panels/ExtensionSidebarPane/setExpression
```
Comment (wbamberg) [inline files/en-us/mozilla/add-ons/webextensions/api/devtools/panels/extensionsidebarpane/setexpression/index.md:5]:
```suggestion
  Mozilla/Add-ons/WebExtensions/API/devtools/panels/ExtensionSidebarPane/setExpression
page-type: webextension-api-function
```

**Review by rebloor — APPROVED** [review]
@wbamberg I've spot checked a number of the pages, and everything looks OK

**Pull request merged**